src/js_network_transport.hpp

```diff
@@ -175,41 +166,58 @@ struct JavaScriptNetworkTransport : public JavaScriptNetworkTransportWrapper<T>
     using Object = js::Object<T>;
     using Value = js::Value<T>;
 
-    JavaScriptNetworkTransport(ContextType ctx)  : JavaScriptNetworkTransportWrapper<T>(ctx) , m_ctx(ctx) {
-        realm_constructor = Value::validated_to_object(m_ctx, Object::get_global(m_ctx, "Realm"));
-        network_transport = Object::get_property(m_ctx, realm_constructor, "_networkTransport");
-    };
+    using SendRequestHandler = void(ContextType m_ctx, const app::Request request, std::function<void(const app::Response)> completion_callback);
 
-    void send_request_to_server(const app::Request request, std::function<void(const app::Response)> completion_callback) override {
-        // Create a response handler
-        ObjectType response_handler = ResponseHandlerClass<T>::create_instance(m_ctx, std::move(completion_callback));
-
-        // Create a JavaScript version of the request
-        ObjectType request_object = Object::create_empty(m_ctx);
-        Object::set_property(m_ctx, request_object, "method", Value::from_string(m_ctx, fromHttpMethod(request.method)));
-        Object::set_property(m_ctx, request_object, "url", Value::from_string(m_ctx, request.url));
-        Object::set_property(m_ctx, request_object, "timeoutMs", Value::from_number(m_ctx, request.timeout_ms));
-        if (!request.body.empty()) {
-            Object::set_property(m_ctx, request_object, "body", Value::from_string(m_ctx, request.body));
-        }
-        ObjectType headers_object = Object::create_empty(m_ctx);
-        for (auto header : request.headers) {
-            Object::set_property(m_ctx, headers_object, header.first, Value::from_string(m_ctx, header.second));
-        }
-        Object::set_property(m_ctx, request_object, "headers", headers_object);
+    JavaScriptNetworkTransport(ContextType ctx) : m_ctx(ctx) {};
 
-        ValueType arguments[] = {
-            request_object,
-            response_handler,
-        };
+    static void init(ContextType ctx) {
+        //this initializes the EventLoopDispatcher on the main JS thread
+        get_send_request_handler();
+    }
 
-        Object::call_method(m_ctx, Value::to_object(m_ctx, network_transport), "fetchWithCallbacks", 2, arguments);
+    void send_request_to_server(const app::Request request, std::function<void(const app::Response)> completion_callback) override {
+        auto send_request_handler = get_send_request_handler();
+        send_request_handler(m_ctx, request, completion_callback);
     }
 
 private:
     ContextType m_ctx;
-    ObjectType realm_constructor;
-    ValueType network_transport;
+
+    static realm::util::EventLoopDispatcher<SendRequestHandler> get_send_request_handler() {
```

Comment:
Should this return `realm::util::EventLoopDispatcher<SendRequestHandler>&` so that it doesn't need to make a copy on every call?

Reply:
yes. good point. will fix